### ISSUE TITLE
loss of content e.g. form field data when moving from one container (context) to another

### ISSUE THREAD

**Issue by grahamperrin** [totallymike/contextPlus#15]
For example: loss of the title of this issue, if I change context before submission. 

Re: https://github.com/mozilla/testpilot-containers/wiki/Moving-between-containers and the nature of Firefox containers, I assume that such losses are unavoidable. So please consider adding an on-screen hint … maybe by default, once per Firefox session, a modal dialogue that will give the user the opportunity to _Cancel_ a move before it's too late. 

Or maybe refrain from closing the tab from the original context, and (after the copy is complete) offer a non-modal dialogue in the new context. 

----

Side note: https://github.com/kesselborn/taborama/issues/8

**Comment by grahamperrin** [comment]
> … refrain from closing the tab from the original context, and (after the copy is complete) offer a non-modal dialogue in the new context.

Maybe think of that (dialogue) as a complement to the <kbd>Control</kbd> key approach. Like, a dialogue if the first change of context during a session is _without_ <kbd>Control</kbd> key modification.

**Comment by grahamperrin** [comment]
> …  I assume that such losses are unavoidable. … 

Was that a false assumption? 

From [What is (and isn't) separated between Containers](https://wiki.mozilla.org/Security/Contextual_Identity_Project/Containers#What_is_.28and_isn.27t.29_separated_between_Containers) under _Security/Contextual Identity Project/Containers - MozillaWiki_: 

> Not separated by Containers (On Purpose):
> 
> - Saved Passwords
> - Saved Search and Form data
> - …

To be honest, I don't know whether (technically) what I type in this form, in this GitHub comment field, is (without clicking the green 'Comment' button) treated as saved form data.

**Comment by totallymike** [comment]
Something like a warning the very first time you switch contexts probably makes sense.

The extension doesn't really inform you that it's straight up destroying the old tab and creating a new one in a different context.

I don't think I'd like to have the warning show up more frequently than that.  I'd get sick of it pretty quickly :)

**Comment by grahamperrin** [comment]
FWIW from [Drawings for Firefox containers](https://discourse.opensourcedesign.net/t/drawing-user-stories-as-cartoons/235/5?u=grahamperrin) under _Drawing user stories as cartoons - Lounge - Open Source Design_ (and via <https://twitter.com/grahamperrin/status/917100338255138818>): 

> … A comical demonstration of something lost through movement, maybe …

**Comment by grahamperrin** [comment]
> <[digression](https://github.com/totallymike/contextPlus/issues/24#issuecomment-335917767)>
> 
> Frankly, I'm not sure what I'd want to do about on-page form data and such. I feel like that's probably out of scope for this add-on since there's just about no way to preserve live data for every page everywhere that I can think of.
> 
> The one thing, I think, that bums me out more than possibly losing half-filled-out Form data, is the back button. When starting this addon, I almost scrapped it all together as not being worth it when I found out I'd lose the back button history!

Yeah. I never expected this extension to do what Firefox can not (or should not). 

On one hand: it's for Mozilla, more than any other group or individual, to manage user expectations – including bum-outs and thoughts of scrap – around its container technology. On the other hand: as [Mozilla positively encourages open design](https://blog.mozilla.org/OpenDesign/), so _anyone_ might come up great ways of conveying what's great, good and not good about Firefox containers. 

@girliemac through your popularity and skills you're invited `:-)` but no pressure.

**Comment by grahamperrin** [comment]
Food for thought. 

The tree on the right is identical to the tree on the left: 

![christmas tree decor ideas](https://user-images.githubusercontent.com/192271/31571025-782fd74a-b083-11e7-9d73-1f28c96bc359.jpg)

The container is different. 

;-)

----

Credit for the original photographs: [Hammers and High Heels: Green and Gold to Go With the Stars & Snow! It's Our 2013 Christmas Tree!](http://hammersandhighheels.blogspot.co.uk/2013/12/of-stars-snow-green-and-gold-its-our.html)

The horizontal flip, and the lazy, flippant pastes, are down to me …
